src/cloudant/database.py

```diff
@@ -24,23 +24,35 @@ def __init__(self, cloudant, database_name, fetch_limit=100):
         self._database_host = cloudant._cloudant_url
         self._database_name = database_name
         self._r_session = cloudant._r_session
-        self._next_startkey = None
         self._fetch_limit = fetch_limit
 
-    _database_url = property(
-            lambda x: posixpath.join(
-                x._database_host,
-                urllib.quote_plus(x._database_name)
-            )
+    @property
+    def database_url(self):
+        return posixpath.join(
+            self._database_host,
+            urllib.quote_plus(self._database_name)
         )
 
+    @property
+    def creds(self):
```

Comment:
I think I would go for a more descriptive name here, like build_creds or something like that, but NBD